Source/WebGPU/WGSL/AST/ASTTypeName.h

```diff
@@ -103,9 +103,20 @@ class ParameterizedTypeName : public TypeName {
         Mat3x4,
         Mat4x2,
         Mat4x3,
-        Mat4x4
+        Mat4x4,
+        Texture1d,
+        Texture2d,
+        Texture2dArray,
+        Texture3d,
+        TextureCube,
+        TextureCubeArray,
+        TextureMultisampled2d,
+        TextureStorage1d,
+        TextureStorage2d,
+        TextureStorage2dArray,
+        TextureStorage3d,
     };
-    static constexpr unsigned NumberOfBaseTypes = 12;
+    static constexpr unsigned NumberOfBaseTypes = 23;
```

Comment:
instead of manually counting, should we just cast the last item, TextureStorage3d, to an `unsigned` int?

Reply:
it may be just as problematic however since it is possible to forget to update it

Reply:
@mwyrzykowski We can't cast `TestStorage3d` because it will be `22` not `23`. We could add `NumberOfTypes` (or equivalent WebKit approved naming) as the last element and cast `Base::NumberOfTypes` to `unsigned`?

Reply:
If we add NumberOfTypes then we will end up causing compiler warnings / errors everywhere NumberOfTypes is not used, right? Which seems not worth it.

I am fine leaving it as-is, we could also cast the value and add 1 to it, but really it is fine as-is too

Reply:
I considered that, but I don't know of any solution that doesn't require manual updates... I can change it to use the last value + 1. I think that's an improvement, since it doesn't require explicitly counting. I'd prefer that over adding the extra entry, since we'll need to handle it in every switch, as @mwyrzykowski mentioned.

Reply:
Oh, we could also use macros, but I think not everyone is a fan 😬